### ISSUE TITLE
Very strange output from the first localization step (TS Map)

### ISSUE THREAD

**Issue by jballet** [fermiPy/fermipy#514]
I think I have come across a serious bug in the localize call.
One source fails to localize, and when I look at the p88y2172_localize.png output (from the initial 
call to TS map) shown below it looks very strange. Instead of the expected peak at the center it shows two peaks on the sides (as if it was periodic???).
![p88y2172_localize](https://user-images.githubusercontent.com/37209477/209376516-b81cb428-4c5e-49f0-b16d-b5bc9d9ddb87.png)
This strange behavior goes away when I change something in the spatial settings. For example I was not able to reproduce it with a larger pixel size or a smaller overall RoI size (either would have reduced the size of the test harness). But it is robust to energy selection (I started with an interval going to 1 TeV), to the XML model (I started with a much more complex one) and also to the components setting (I started with 4 components for each PSF event type).
I simplified the calls and prepared a test harness at ftp://ftp.cea.fr/incoming/y2k01/jbbglast/fermipy_fail.tgz
It is 75 MB so I prefer not attaching it here directly. It should stay there for at least a week. It is self contained except for the big gll_iem_v07.fits file. Just follow the fermipy_fail.py script. It takes a few minutes to run (the longest part is the full TS map, which can be skipped since it works all right).
The strange p88y2172_localize.png is included, as well as the (perfectly normal) full TS map.
I think this is not very rare because it happened among the first 10 RoIs I tested.
I have originally detected that on v1.1.6, which is the version installed on my PC farm, but I checked v1.2 behaves in the same manner.

**Comment by henrikef** [comment]
@jballet Sorry, I did not have time to look at this before due to the holidays and the AAS meeting. This looks indeed strange. So you say the full TS map looks ok, but then inside the localization you get two peaks? 
Unfortunately, I am not able to download your test file - I try to connect as guest and get a message that the server is unavailable. Would you be able to provide this file in a different way?

**Comment by jballet** [comment]
I have put it back at the same place. Its lifetime is only a week so after 3 weeks it is not surprising that you could not see it.

**Comment by ndilalla** [comment]
Hi @jballet, thanks for posting the example. We were able to reproduce your issue. We discovered that setting the coordinate system to celestial (CEL) leads to the right map in output:
![p88y2172_localize](https://user-images.githubusercontent.com/16223421/213307870-bb1eb168-5174-4c03-bd91-d118b276a345.png)
We need to investigate a bit on what is causing this issue as it seems a bug somewhere in the conversion between galactic and celestial coordinates.

**Comment by jballet** [comment]
Do you mean that you started anew from the FT1 file and recreated the count cubes? If so, I think you do not see the problem because the pixel configuration is different (this problem does not occur all the time), not because you are in celestial coordinates. Is there actually a conversion between Galactic and celestial coordinates inside the localization loop? I don't see why there should be.
I think you should concentrate instead on what happens precisely in the case that fails.

**Comment by ndilalla** [comment]
No, we didn't start with a new FT1 file, we used the files you provided and simply set the output coordinate system to celestial in the configuration file. We believe there is a 1 pixel shift somewhere using galactic coordinates that causes the pattern you observe in the TS map.

**Comment by jballet** [comment]
Formally it works like that, yes, but the result is wrong. I think this is simply because it is inconsistent to work with CEL in the configuration file but existing source maps in GAL coordinates. For information, this is what the localization file p88y2172_localize.png should look like in CEL, when the source maps are recreated in CEL from the FT1 file.
I still doubt that Galactic coordinates have anything to d
![p88y2172_localize](https://user-images.githubusercontent.com/37209477/213748449-49887a4f-8793-4255-b0a1-c2ec2b8cb1fb.png)
o with this issue.

**Comment by ndilalla** [comment]
Hi @jballet I've committed a fix for the issue on the branch [fix_loc_issue_514](https://github.com/fermiPy/fermipy/tree/fix_loc_issue_514). The issue with the two-peaked TS map should now be solved. Could you please try again and see if the output is consistent with what you expected? Thanks

**Comment by jballet** [comment]
Hi Niccolo,
Can you give step-by-step instructions explaining how to get hold of that development version?
We don't have a development environment for fermipy here.

**Comment by henrikef** [comment]
@jballet to install directly from a branch, you should be able to use 

```
pip install -U git+https://github.com/fermiPy/fermipy@fix_loc_issue_514
```

From inside an environment that already has the fermitools etc.

**Comment by ndilalla** [comment]
Quick update: if I run the analysis I get this plot in output:
![p88y2172_localize_peak](https://user-images.githubusercontent.com/16223421/214976763-089bd8d5-89dc-49b2-b3a6-1b2a13248244.png)
The white square region for the refined localization search is completely off with respect to the position of the TS maximum and consequently the localization fit fails. Does anyone has experience this?

**Comment by henrikef** [comment]
What's the other plot look like (from the first localization search)?

**Comment by ndilalla** [comment]
Sorry, I meant to upload that one! 
![p88y2172_localize](https://user-images.githubusercontent.com/16223421/214999528-5649351e-a5a2-459d-ac93-7682015e36b0.png)

**Comment by jballet** [comment]
According to the documentation, the square used for the second step is defined from the approximate localization in the first step (which is the green cross and the green ellipse). What needs to be understood is why this green cross is completely off the maximum.

**Comment by jballet** [comment]
I have just tried the development version here and now the localization works well in GAL mode.
However it fails in CEL mode with an offset TS map peak similar to what you report above.
I noted an unusual warning
/feynman/work/projets/dapglast/envs/fermi-2.2.0G/lib/python3.9/site-packages/scipy/optimize/_lsq/common.py:147: RuntimeWarning: invalid value encountered in scalar power
  alpha = max(0.001 * alpha_upper, (alpha_lower * alpha_upper)**0.5)

**Comment by jballet** [comment]
The development version provided by Niccolo on April 13 locates that source at the proper place in both GAL and CEL coordinates. I will now test it on more sources.
The localize_peak.png plot, however, has the legend r
![p88y2172_localize_peak](https://user-images.githubusercontent.com/37209477/235158806-bb37345d-e887-48ae-8271-eef7e2c2997e.png)
ight at the middle of the plot. This is probably not intended (it goes into a corner in CEL, or if I change the pixel size).

**Comment by jballet** [comment]
I have exercised it quite a bit. The fit provided in patch 514 seems to have solved this problem.

**Comment by jballet** [comment]
I have tried lauching a full run with many RoIs in CEL coordinates. I found that fermipy 1.2.0 fails in just about the same fraction of sources (about 10%) as it does in GAL coordinates, with the same weird localization maps (not the same sources as in GAL coords in general, because of the sensitivity to pixellization).
So I come back to my original idea that the problem has nothing to do with GAL or CEL.
The patch_514 version provided by Niccolo in April avoids that error in CEL as well, but then fails in a similar fraction of sources (not the same) on the "localization contour" error ([issue 548](https://github.com/fermiPy/fermipy/issues/548).
